### ISSUE TITLE
[MAYA-102098] UFE : File save / re-open duplicated prim is gone

### ISSUE THREAD

**Issue by cfmoore007** [Autodesk/maya-usd#269]
**Describe the bug**

Duplicating prims now seems to work in the latest maya beta.  However, once you save and re-open the scene, it seems the duplicate is lost.
If I then select the same prim ( I duplicated earlier ), then press 'ctrl-d' to duplicate again, it magically re-appears where I last moved it.

This was tested w/ the 3_Spheres example and the Pixar Kitchen Set

It appears the duplicate is just described as an over w/ an xformOp.  It does not reference it's source prim.

**Steps to reproduce**
1. open kitchen set
2. select one of the papers on the table
3. ctrl-d
4. move paper
5. save file
6. re-open file, see the paper has now disappeared

**Expected behavior**
Results stay consistent - duplicated prim doesn't disappear on file-re-open.

**Attachments**
3_Spheres Example : 
[3_Spheres_Example.tar.gz](https://github.com/Autodesk/maya-usd/files/4181967/3_Spheres_Example.tar.gz)

**Initial Duplicate & Move** ( works )
![duplicatedPrim](https://user-images.githubusercontent.com/2152250/74169939-1e458080-4bfa-11ea-8126-325446c8b018.png)

**After File Re-open** ( notice it's gone )
![duplicatedPrim_lostAfterSaveAndReopen](https://user-images.githubusercontent.com/2152250/74169947-20a7da80-4bfa-11ea-89e6-1ca0e375fc41.png)

**After File Re-open, ctrl+d** ( magically reappears )
![duplicatedPrim_lostAfterSaveAndReopen_duplicateAgain](https://user-images.githubusercontent.com/2152250/74169954-243b6180-4bfa-11ea-8f02-7605a8d3b1d6.png)

**Specs (if applicable):**
- OS & version [Windows 10]
- Compiler & version [VS2017]
- Maya version [Maya Beta and 2020.0 tested]
- Maya USD commit SHA [dev at a29704f with PR#228 merged in]
- Pixar USD commit SHA [master v20.02 compiled against rman23]

**Additional context**

**Comment by santosg87** [comment]
thanks for bringing this up! 
we have this issue logged internally, and we are looking into it currently. :)

it's logged as MAYA-102098

**Comment by ppt-adsk** [comment]
Hey George, to my knowledge, we don’t save out any USD layer currently, so any USD change you make will be lost.  I’d try it out with any mutating USD operation (rename, translate, delete, whatever).

I think we need to start implementing USD layer workflows to get this going.

Cheers,

                                Pierre

**Comment by cfmoore007** [comment]
If you modify existing prims, save and re-open, the session layer is still kept in tact.
You can try this with the 3_Spheres example in the description.
There's just no definition of 'newly' created prims ( ie: if it's a reference or instance of the source of the duplicate )

**Comment by cfmoore007** [comment]
I should also clarify, I was testing w/ the AL proxy node.  Not the Maya USD proxyShape

**Comment by santosg87** [comment]
@cfmoore007 this seems to be only during the session (which is the behavior of issue #83 ). if you start a new session of Maya and try opening that same scene, your changes will be lost.

**Comment by cfmoore007** [comment]
Right.  In the AL plugin case, what I'm seeing written to the session layer is incomplete :
`
select -r AL_usdmaya_ProxyShape ;
# get selected proxy shape
proxyShape = cmds.ls(selection=True)[0] +"Shape"

# get session layer identifier ( identifiers=True is required for this )
sessionLayer = cmds.AL_usdmaya_LayerGetLayers(proxy=proxyShape,sessionLayer=True,identifiers=True)
 
# query session layer at current time sample
print cmds.AL_usdmaya_LayerSave(sessionLayer,proxy=proxyShape,string=True)
#usda 1.0

over "pSphere4"
{
    double3 xformOp:translate = (0, 0, -5.7115959338931805)
    uniform token[] xformOpOrder = ["xformOp:translate"]
}
`
over "pSphere4" only has it's xformOp defined, not that it's a reference, or instance to it's source prim.

As for the Maya proxyShape, I'm unsure how to query it's session layer to double check that.

**Comment by cfmoore007** [comment]
This issue still persists in the latest beta.

**Comment by santosg87** [comment]
@cfmoore007, this one should be fixed now - are you able to validate? :)

**Comment by santosg87** [comment]
nvm. it still happens :(

**Comment by kxl-adsk** [comment]
@cfmoore007 Can you test this workflow with adsk proxy and PR #1152 ?

**Comment by cfmoore007** [comment]
Confirmed, this now works.
![duplicateStays](https://user-images.githubusercontent.com/2152250/107222127-8b911a80-69e2-11eb-8a22-4eb45c3202c7.png)

Note : for duplicating prims currently to work, it seems the edit target must be set to the source .usd where it was originally defined.  In my case with 3 spheres example, I set the edit target to that layer and made my duplicate there.  I then proceeded to save my maya file and saved edits to .ma ( and not back to my original ).  This is why you see in my example the layer editor says the stage is still dirty and needs to be saved.
